src/qibolab/platforms/abstract.py

```diff
@@ -152,6 +163,135 @@ def reload_settings(self):
+                    True
+
+                # resonator_punchout_attenuation
+                elif par == "bare_resonator_frequency":
+                    freq = int(value * 1e9)
+                    self.qubits[qubit].bare_resonator_frequency = freq
+                    self.current_config["characterization"]["single_qubit"][qubit]["bare_resonator_frequency"] = freq
+
+                # resonator_spectroscopy_flux / qubit_spectroscopy_flux
+                elif par == "sweetspot":
+                    sweetspot = float(value)
+                    self.qubits[qubit].sweetspot = sweetspot
+                    self.current_config["characterization"]["single_qubit"][qubit]["sweetspot"] = sweetspot
+
+                # qubit_spectroscopy / qubit_spectroscopy_flux / ramsey
+                elif par == "drive_frequency":
+                    freq = int(value * 1e9)
+                    self.native_single_qubit_gates[qubit]["RX"]["frequency"] = freq
+                    self.current_config["native_gates"]["single_qubit"][qubit]["RX"]["frequency"] = freq
+
+                    self.qubits[qubit].drive_frequency = freq
+                    self.current_config["characterization"]["single_qubit"][qubit]["drive_frequency"] = freq
+
+                elif "amplitude" in par:
+                    amplitude = float(value)
+                    # resonator_spectroscopy
+                    if par == "readout_amplitude" and not math.isnan(amplitude):
+                        self.native_single_qubit_gates[qubit]["MZ"]["amplitude"] = amplitude
+                        self.current_config["native_gates"]["single_qubit"][qubit]["MZ"]["amplitude"] = amplitude
+
+                    # rabi_amplitude / flipping
+                    if par == "drive_amplitude" or par == "amplitudes":
+                        self.native_single_qubit_gates[qubit]["RX"]["amplitude"] = amplitude
+                        self.current_config["native_gates"]["single_qubit"][qubit]["RX"]["amplitude"] = amplitude
+                        self.current_config["characterization"]["single_qubit"][qubit]["pi_pulse_amplitude"] = amplitude
+
+                # rabi_duration
+                elif par == "drive_length":
+                    duration = int(value)
+                    self.native_single_qubit_gates[qubit]["RX"]["duration"] = duration
+                    self.current_config["native_gates"]["single_qubit"][qubit]["RX"]["duration"] = duration
+
+                # ramsey / spin_echo
+                elif par == "t2" or par == "t2_spin_echo":
+                    t2 = float(value)
+                    self.qubits[qubit].T2 = t2
+                    self.current_config["characterization"]["single_qubit"][qubit]["T2"] = t2
+
+                # t1
+                elif par == "t1":
+                    t1 = float(value)
+                    self.qubits[qubit].T1 = t1
+                    self.current_config["characterization"]["single_qubit"][qubit]["T1"] = t1
```

Comment:
In this example I was updating the self.current_config because then, in the runcard dump, I could pass self.current_config (or self.settings when we remove self.current_config, to avoid duplications) and it is for sure, totally updated. Otherwise, we need to recreate the runcard from the platform class attributes (dictionaries or data structures in the future) and, at this point, as we need the self.settings dictionary to be passed to the setup functions, I decided to update the fitted values in 2 different places.